\begin{abstract}
% Abstract length should not exceed 250 words
Does the probability of evolution producing a trait vary across space based on local conditions? Biological data demonstrate substantial spatial variation in some evolutionary processes, such as diversification rate. However, the evolution of novel traits has not been addressed in this context, owing in part to the challenges of conducting the necessary experiments with natural organisms. We conduct \textit{in silico} experiments with the Avida Digital Evolution Platform to address this question. We created eight different spatially heterogeneous environments and ran 100 replicates in each.  Within each environment, we examined the distribution of locations where nine different focal traits first evolved. Using spatial statistics methods, we identified regions within each environment that had significantly elevated probabilities of containing the first organism with a given trait (i.e. hotspots of evolutionary potential). Having demonstrated the presence of many such hotspots, we explored three potential mechanisms that could drive the formation of these patterns: proximity of specific resources, variation in local diversity, and variation in the sequence of locations the members of an evolutionary lineage occupy. Resource proximity and local diversity appear to have minimal explanatory power. Lineage paths through space, however, show some promising preliminary trends. If we can understand the processes that create evolutionary hotspots, we will be able to craft environments that are more effective at evolving targeted traits. This capability would be useful both to evolutionary computation, and to efforts to guide biological evolution.

\end{abstract}
\section{Introduction}

Understanding the evolutionary processes that facilitate the evolution of complex traits is at the heart of many open questions in artificial life, evolutionary computation, and evolutionary biology. Numerous studies have suggested that spatially complex environments can drive the evolution of complex traits, but it is unknown whether this result is a global pattern that is only visible when summarizing results across an entire environment, 
or whether it is driven by local spatial dynamics within certain sub-regions of the environment. If sub-regions do exist that disproportionately promote the evolution of specific adaptive traits (hereafter referred to as \textit{evolutionary hotspots}), they would be a valuable phenomenon to understand. If we could determine what factors make a region an evolutionary hotspot, we could likely design environments that dramatically increase the likelihood of complex traits evolving. This ability would enable us to build more interesting artificial life systems, more effective evolutionary algorithms, and, ideally, biological environments that promote a desirable evolutionary outcome.

Where should we expect to find evolutionary hotspots? Instinctively, we might expect a trait to arise only in regions where it is beneficial. However, selection can only favor traits once they appear, not promote that initial appearance, so other aspects of the environment must play a critical role.  At the other extreme, hotspots seem unlikely in a homogeneous environment with no spatial structure.
However, few environments in nature are truly homogeneous, and almost all have some form of spatial structure. Even among digital systems, once we consider the fact that any kind of interaction between neighboring individuals introduces spatial heterogeneity, surprisingly few are truly spatially homogeneous. Thus, understanding if and how evolutionary potential varies across space could be important for understanding evolutionary dynamics in a variety of systems.

Why should we expect evolutionary hotspots to exist? To better understand this idea, it is useful to consider Wright`s fitness landscape metaphor \citep{wright_roles_1932}. At its simplest, a fitness landscape is a three-dimensional surface representing two traits (x and y) with varying values, and the fitness of an organism having a given combination of those trait values (z). When there are gradients in this space, we expect populations to climb to a fitness peak over time. Typically, populations will then get stuck at the peak of a local gradient, regardless of whether it is the highest location in the landscape. As such, the topography of the local fitness landscape plays a pivotal role in determining what traits will evolve in a population. A wide variety of factors can influence what part of a fitness landscape a population ends up in at a given point in time \citep{wilke_evolution_2001}. We predict that spatial variation across the environment is one such factor. A spatially heterogeneous environment can be conceptualized as a network of different fitness landscapes, one for each location in space. However, in a dynamic, interdependent ecological community, these fitness landscapes shift over time. Some local landscapes will tend to guide portions of the population toward regions of mutational space from which new traits are more easily accessible.

Of course, this network of fitness landscapes may not guide evolving populations in a consistent direction. While it is possible that landscapes could align to perfectly guide a lineage through the most challenging steps in the evolution of a complex trait, this condition seems unlikely. We might just as easily expect that the lack of a constant landscape will only serve to increase stochasticity, rather than having a predictable effect. Perhaps the fitness landscapes in adjacent locations even work at cross-purposes, with each guiding the population in opposite directions. Given this uncertainty, the primary goal of this paper will be to determine whether hotspots actually occur with sufficient frequency and strength to be worthy of further consideration. Addressing this question requires vast numbers of repetitions of experimental evolution in complex spatial environments. Conducting such an experiment in a wet lab system would not be worth this immense effort until additional theory has been developed. Once we have generated concrete hypotheses about what we would expect to see in biology, more targeted follow-up experiments in a wet lab system may be worthwhile. Here, we lay the groundwork for such theory by performing experiments in a digital system, where we will be able to precisely control the spatial distribution of environmental conditions and measure the drivers behind any spatial variation in evolutionary potential that we detect.

We focus this paper on a relatively simple environment in order to avoid adding unnecessary complications to an already complex question. To this end, we use sessile, asexual  organisms in an unchanging abiotic environment where different sets of traits are advantageous in different regions. Importantly, these traits are related to each other. This set-up is roughly analogous to a wide variety of scenarios within biology, ranging from bacteria surviving on a surface treated with a range of antibiotics to plants adapting to survive on different soil types within a region. 
% * <mjwiser@gmail.com> 2017-04-12T16:22:01.026Z:
% 
% >  ranging from bacteria surviving on a surface treated with various antibiotics  to plants adapting to survive on different soil types within a region. 
% 
% I feel like you want some citations here.
% 
% ^ <emilyldolson@gmail.com> 2017-04-13T20:14:50.442Z:
% 
% Citations indicating that those systems exist? Or have those properties? The comparison isn't really established to be accurate - it's an assertion I'm making.. (or maybe what you're saying is I shouldn't make such an assertion without citations to back it up :) )
% 
% ^.
%While these experimental settings minimize temporal stochasticity, it is important to keep in mind that they do not eliminate it. The selective pressures in our experiments ultimately come from competition with neighboring organisms. As such, while the fitness landscape that an organism experiences is primarily determined by its abiotic environment, this landscape is deformed slightly by the competitive ability of neighboring organisms.
%Hmm, actually, now I'm doubting this. There isn't actually density dependence for a given task. When I wrote this, I was thinking about organisms on the edge of one patch competing with organisms on the edge of a different patch. But I think that can actually all be encapsulated in the abiotic fitness landscape. You just need to use expected value of fitness and consider the fitness of your offspring when it lands in a different patch in proportion to the probability of it landing there. %MJW: I agree with your reasoning.  When you run an expected value calculation involving the probability of your offspring being born into a different environment, that's not bad.


\section{Background}

There is a great deal of evidence from biology that evolutionary processes are not consistent across space. For the most part, biologists have studied patterns of diversification rather than the patterns of novel trait evolution that we address here. These processes have important distinctions from each other; diversification usually requires either physical separation of sub-populations or conflicting selection pressures, whereas novel trait evolution does not. Moreover, they are maximized by different conditions \citep{walker_evolutionary_2012}. Nevertheless, diversification and novel trait evolution are closely related, as diversity is thought to increase evolutionary potential \citep{rouzic_evolutionary_2008}. The most well-studied biological pattern pointing to spatial variation in evolutionary processes is the latitudinal diversity gradient: a consistent pattern of high biodiversity near the tropics, and progressively less as you move toward the poles \citep{hillebrand_generality_2004}. While the impossibility of performing controlled experiments on planetary-scale patterns of evolution has made untangling the drivers of this pattern challenging, much progress has been made. Most importantly for the purposes of this paper, there is mounting evidence that diversification rates are higher in the tropics \citep{mittelbach_evolution_2007}. The drivers of this variation are still being debated, but many hypothesized drivers are related to spatial dynamics within the environment, most notably the idea that the tropics may contain more environmental gradients, barriers to gene flow, and other spatial properties that promote speciation \citep{moritz_diversification_2000}. Although these dynamics are more complex than those addressed in this paper, particularly given that many require sexual reproduction in order to be relevant, they suggest that spatial variation in evolutionary processes is a phenomenon that has broad relevance.
% MJW:
% >  these processes have important distinctions from each other
% Such as?
% 
% ELD: I would love to know the answer to that question.
%
% CAO: Should we change it to reflect that novel trait evolution focuses on specific traits arising, while diversity is about any new trait (or merely a new combination of traits) arising?
%ELD: Better?

In related work, conservation biologists have sought to locate regions with an elevated intensity of evolutionary processes. This research stemmed from an increasing recognition that preserving the processes that generate diversity is at least as important as preserving existing diversity \citep{frankel_genetic_1974,moritz_strategies_2002,ferriere_evolutionary_2004}. In accordance with this motivation, these studies tend to focus on small regions of land, making them
more directly comparable to the research presented here. At a global scale, certain regions simply lack the genetic background for a given trait to have a chance of evolving (e.g. a novel heat tolerance mechanism seems unlikely to evolve in the arctic). Our question centers on more localized regions that are spatially heterogeneous with regard to the extent to which various related traits are useful. The results from conservation biology suggest that variation in evolutionary potential on such a local scale is plausible; even within relatively small regions, these studies have demonstrated substantial spatial variation in phylogenetic diversity \citep{forest_preserving_2007,vandergast_are_2008}. 

%I don't know how I feel about this paragraph. I feel like I should be saying something somewhere about tangible effects that different spatial patterns can have on evolution. But that makes for a lot of disjointed facts to summarize. So it feels kind of awkward, and I'm not sure if these are the most important spatial dynamics to be mentioning.
%
% CAO: I thought this paragraph held together surprisingly well (leading through how islands are used as a test case and how they can be generalized), but I might have been biased by your intro and expecting it to be bad.
%ELD: Actually, now that I re-read it, I'm more okay with it (although I might be biased by being tired of messing with it)
Most of the processes that are hypothesized to be involved in the formation of hotspots of phylogenetic diversity are based on insights from simpler systems, most notably island groups. Islands represent clearly delineated spatial patterns and processes, making them an excellent context in which to build up insights about how eco-evolutionary dynamics will play out across space. For example, islands are known to promote diversification via adaptive radiation \citep{losos_adaptive_2010}, and the formation of new islands within a system has substantial impacts on the assembly of ecological communities over evolutionary time \citep{gillespie_community_2004}. Island systems are also a particularly pronounced example of variation in spatial structure. Spatial population structure has a profound effect on how evolution will proceed; more structured populations (i.e. those with more constraints on where offspring end up relative to their parents) are generally more diverse and more likely to reach the highest peak in a fitness landscape \citep{tomassini_spatially_2005,nahum_tortoisehare_2015}. The theory of island biogeography makes various shorter timescale predictions \citep{macarthur_theory_1967}. More recently, biologists have begun to translate theory developed for island systems to more continuous land masses, often by treating regions of a species' preferred habitat as "islands" (also called patches) within a matrix of less suitable habitat \citep{forman_land_1995}. Many have argued this approach is likely an oversimplification \citep{mcgarigal_surface_2009,franklin_importance_2009}. A better understanding of the relationship between the dynamics observed in island systems and the dynamics that occur in more continuously varying environments will likely be important to understanding the spatial evolutionary dynamics that create evolutionary hotspots.  


\section{Methods}

\subsection{Study System}
As previously discussed, addressing this question requires a system with a relatively complex abiotic fitness landscape. The Avida Digital Evolution Platform is an obvious choice for such a system \citep{ofria_avida:_2004}. Not only does it easily allow for the creation of such fitness landscapes, it also includes a variety of tools for analyzing them. Avida is a world of self-replicating linear computer programs competing for space. Mutations are introduced during the self-replication process, resulting in evolution via natural selection, with programs that can copy themselves faster having higher fitness. Improvements in fitness can either come from optimizations to the code of the genome (increasing its efficiency) or from evolving to take in resources by performing boolean logic operations. Resources give programs extra CPU cycles, allowing them to copy themselves faster than competitors. Each of the nine two-input Boolean logic functions allows organisms to use a different resource, with more complex functions corresponding to more valuable resources. Here, we will use the ability to perform these functions as the traits of interest, and define organism phenotypes as the set of traits they possess.

Programs in Avida inhabit a 60 x 60 grid and offspring are placed in cells adjacent to their parents. Results presented here use non-toroidal (bounded) grids, but otherwise identical experiments on a toroidal grids produced similar results.  We create heterogeneous environments by placing different resources in each cell. For a previous experiment, % @CAO: CITE!!! %@ELD: But since it got rejected there currently isn't anything to cite... :(
we created 500 environments by placing a random number of randomly-sized circular patches of resources in random locations across the environment. This technique produced a variety of environments while maintaining a correlation between the resources present in one cell and the resources present in its neighbors. For the purposes of this project, we have arbitrarily selected eight environments for further study. % @CAO: How were they chosen?  Should it say "randomly selected"?  Or was there an effort for a range of environments, at which point you should try to state how you did that. %@ELD: If I'd been planning ahead better when I first started messing around with this project, I would have done it randomly. Instead, I just selected a sequence of 8 environment files that were adjacent to each other in my file browser. So we could say "arbitrarily selected." 
In each environment, we performed 100 replicate runs of Avida. Each replicate was allowed to run for 100,000 updates (approximately 2,000 generations). We also performed 100 replicate runs of a homogeneous control condition where all resources were globally available. From these results, we extracted the coordinates of the grid cell in which each trait first appeared within each run. While a trait may evolve independently multiple times within a run of Avida, including these subsequent evolutions would introduce various biases into the dataset (especially when traits were lost and then re-evolved). Thus, we consider only the first organism ever to display a given trait within a replicate, before selection has an opportunity to play a role.

Our extracted (x,y) coordinates translate into 72 separate point patterns (see Figure \ref{stats}A): one for each of the nine different traits in each of the eight environments. Each pattern contains a maximum of 100 points. Many patterns contain fewer points, as it is not guaranteed that every trait will evolve in every replicate. Because each replicate is completely independent of each other replicate, the only relationship between points in the same pattern is that they evolved in the same environment.

\subsection{Statistical Approach}

\begin{figure}
\begin{center}
\includegraphics[width=3.35in]{figs/statsfig_horziontal_1col.png}
\caption{Hotspot location technique. A) An example point pattern, generated from 100 runs of Avida within a single environment. This pattern corresponds to the most complex trait, EQU. Each point represents the location in space of the first organism within one of the 100 runs that was able to perform EQU. B) Kernel intensity map generated from the point pattern in part A. C) Regions within that kernel intensity map that are substantially more intense than observed in the Monte Carlo simulations (i.e. the hotspot).}
\label{stats}
\end{center}
\end{figure}
\begin{figure}[t]
\begin{center}
\includegraphics[width=2.5in]{figs/9_50013_k-hat.png}
\caption{Results of performing the test for spatial clumping on the point pattern from Figure \ref{stats}A. The observed values of the test statistic are well above the values observed in the Monte Carlo simulations (shaded area), so we conclude that there is clumping in this pattern. The red line indicates the theoretically derived expected value for a random pattern.}. 
\label{khat}
\end{center}
\end{figure}

To determine whether or not different regions have different evolutionary potential, we need to determine whether the point pattern is significantly different from complete spatial randomness. To make it possible to appropriately correct our statistics for multiple comparisons, we performed the test for spatial randomness in two steps: 1) a test to see whether each pattern, as a whole, was random, and 2) a follow-up test to determine which regions of non-random patterns had significantly more points than expected by chance. 

For the overall randomness test, we calculated a test statistic for various distances, $r$. For each point in the pattern, we counted the number of other points falling within $r$ distance units of it. We then averaged this number across all points to get our test statistic for that value of $r$. For each point pattern, we generated a range of expected results under the null hypothesis of complete spatial randomness by running 100 Monte Carlo simulations where we randomly placed the number of points in our actual data across an area of the same size. If the value of the test statistic calculated from our observed data was greater than the highest value from the Monte Carlo simulations for more than 5\% of the $r$ values, we concluded that the observed pattern was substantially more clumped than we would expect to see by chance (see Figure \ref{khat}). A test statistic below the range of those observed for the Monte Carlo simulations would suggest that the observed data was more uniform than we would expect by chance; such a result would be unexpected for our experiments. Note for those familiar with spatial statistics: this statistical test is functionally equivalent to Ripley's K function \citep{ripley_mapped_1981}. However, since Ripley's K function assumes first-order stationarity and our data is exclusively the result of first-order trends, we do not expect to see the same patterns traditionally observed in Ripley's K function.

% Need to play more with the colors on this figure and make the legend have task names, not numbers (people at ECAL aren't going to be scared by the task names like biologists sometimes are)
% @CAO: Agreed!  I'm also tempted to suggest making this bigger if we have the room.  These are very pretty graphs.  On the other hand, I think they'll be big enough as-is for reviews to see as much detail as they want.	
%@ELD: Like make it 2x4 instead of 4x2 so it takes up basically the whole page? I think I'd prioritize making the paths figure at the end bigger, depending on how much space we have. (now that I've messed with Figure 5, I'm assuming we don't have space)
%@ELD: Are these colors better?
\begin{figure*}[t]
\begin{center}
%\includegraphics[width=2.1in]{50065_all_hotspots.png}
\includegraphics[width=7in]{figs/50065_all_hotspots.png}
\caption{Hotspots across all traits and all environments. Each panel represents a different environment. Background colors represent the different combinations of resources that are present in each location; each color represents a different combination, with colors assigned such that higher hue values %@CAO: Are readers going to know what higher hue values are? I assume they're ordered that same as the rainbow? @ELD: I would really hope so, but maybe not. They are ordered the same as the rainbow, but I'm not sure how to see that succinctly (normally I'd say "cooler colors", but that magenta isn't very cool)
roughly correspond to more complex environments (either in number of resources or complexity of traits associated with resources). Each polygon color represents the outline of hotspots for the trait specified in the legend. Note that only hotspots significant at an alpha-level of .00001 are shown. This conservative value was necessary to be certain that hotspots are present despite the large number of hypothesis tests performed. However, in practice, it likely translates to an underestimation of the true size of many of the hotspots.}
\label{hotspots}
\end{center}
\end{figure*}

For those point patterns that were significantly more clumped than we would expect by chance, we performed a follow-up test to determine the location of the hot spot within the environment. We calculated a kernel intensity surface (essentially a heat map of point density across the environment) for each of these point patterns (see Figure \ref{stats}B). Then, in order to determine whether or not the patterns within the data were significantly more intense than we would expect from a random process, we performed a Monte Carlo hypothesis test with 100,000 simulations. We simulated patterns generated under the assumption of complete spatial randomness (our null hypothesis) by randomly selecting points from a uniform 60x60 lattice grid approximating the lattice of cells in Avida.  For each point pattern, we selected the number of points observed in the original pattern. Kernel intensity values were created for each of these simulations. For each cell in the grid, we compared the experimentally-derived intensity to the distribution of simulated intensity values. Cells in which our observed value was higher than 99.999\% of the simulated values were labeled as hotspots of evolutionary potential for a given trait. This significance threshold ($p < .00001$) was selected to ensure all results are still significant after a sequential Bonferroni correction for multiple comparisons across all 3600 grid cells.

\subsection{Code and data availability}
Avida is open source and freely available at \href{https://github.com/devosoft/avida}{https://github.com/devosoft/avida}. Scripts used to generate environment files are available via Zenodo with DOI 10.5281/zenodo.162981. All statistical analyses were performed using the spatstat library, the qqtest library, and the R Statistical Computing Language \citep{baddeley_spatial_2015,oldford_qqtest:_2016, team_r:_2013}. Data and scripts used in this project are available at \href{https://github.com/emilydolson/evo_in_space}{https://github.com/emilydolson/evo\_in\_space}.

\section{Results and Discussion}
\subsection{Existence of evolutionary hotspots}
Out of the 72 point patterns, 43 differed significantly from random. All 43 of these non-random patterns showed significant clumping of points. For 40 of these 43 non-random patterns, the kernel density map had at least one region of higher intensity than we would expect to see by chance (hotspot) (see Figure \ref{hotspots}). It is possible that the three patterns without hotspots instead had areas of significantly decreased evolutionary potential. Alternatively, these patterns may contain hotspots that were too weak to be detected by our highly conservative threshold for significance. Every test environment had hotspots for at least two traits, while none of the point patterns from the spatially homogeneous control displayed significant clumping, suggesting that these hotspots are a product of the spatially heterogeneous environment. These results provide clear evidence that hotspots of evolutionary potential not only exist but are relatively common; some regions of the environment promote the evolution of new traits to a greater extent than others.

In most environments, there appears to be strong overlap in hotspots for the evolution of different traits (see Figure \ref{hotspots}). However, there are some environments in which this is not the case, and even the environments with the neatest overlap have some traits with non-overlapping hotspots. Some amount of overlap is to be expected, for three reasons: 1) If any trait is easier to evolve in a given region, then lineages living in that region are more likely to be successful as a result of having evolved the trait, giving them an older evolutionary history, 2) many traits serve as building blocks for other traits, so possessing some traits is an indicator that a lineage is in a part of the fitness landscape from which other traits are easier to reach, and 3) some traits are close to each other within the fitness landscape (for example, the two most complex traits, XOR and EQU, which often have hotspots in nearly identical locations). The fact that the overlap is not universal suggests that the local environment has effects on the likelihood of evolving a trait that go above and beyond the ability of lineages to perform other traits.  

\subsection{Potential drivers of hotspots}
Now that we have established that a spatial pattern exists in the locations where traits first evolve, the obvious next question is: what causes this pattern? Since all traits have potential to serve as building blocks for each other, the simplest explanation would be that locations with more resources have higher evolutionary potential. Note that because we are considering the first mutation to ever confer each trait, the resource for that trait cannot exert selective pressure.

To test this explanation, we performed a multi-variate linear regression attempting to predict the value of each position in the kernel intensity map from a series of boolean variables indicating which resources were present in a cell. We tried such regressions on various subsets of the data: a single trait within a single environment, a single trait across environments, and all traits across all environments. Models built from data on a single trait in a single environment varied wildly in the percent of variation that they explained, but many explained a high percentage ($R^{2}$ values ranged from .11 to .95). However, very little (if any) of this explanatory power generalizes across environments. Resources that increase evolutionary potential for a given trait in one environment (as indicated by a regression coefficient significantly greater than 0) will decrease it in a different environment (as indication by a regression coefficient significantly less than 0). Linear models calculated across environments were unable to reconcile these differing patterns, explaining minimal variation in evolutionary potential ($R^{2}$ values ranged from .03 to .13). A similar set of models in which the predictor variables were the distance to each resource was similarly ineffective. While high degrees of spatial autocorrelation in the kernel intensity maps mean that the assumption of independence is violated for these linear models, such a reduction in the degrees of freedom should only inflate the apparent variation explained. The failure of this simple mechanism to explain our observed results suggests that there are more complex processes driving these effects.
%Should I show a regression table(s) here?
% @CAO: If you have the room, it's a good addition, but most of the audience (sadly) won't really understand it beyond "wow, she actually knows statistics."  If this were a bio paper, I'd say we should even make room for it.
%@ELD: That is sad, but probably true. Since space is pretty iffy at this point, I'll leave it out for now.

Many hotspots appear to overlap with borders between different combinations of resources. Such regions likely have elevated local diversity, as different phenotypes will be better adapted to the different regions. As discussed previously (see Background), there is reason to believe that diversity leads to increased evolutionary potential. Thus, variation in local diversity is another possible driver of evolutionary hotspots. To test this hypothesis, we calculated local diversity across the environment for time points immediately before a novel trait first appeared. Specifically, we measured local diversity as the Shannon entropy of phenotypes within the 25 cells in the 5x5 patch around a given focal cell. We performed this calculation for each cell in the environment at a given time point, producing a map of local diversities. Within each of these maps, we calculated a percentile for each of the 3600 local diversity values, based on the proportion of sites that had a lower diversity.
% @CAO: "Same rank" isn't always clear.  Do they all count as if they were just one organism?  Or is spece left for the whole group?  Is a group rank in the best, worst, or medium positions of the spots it holds?  One simpler way of phrasing this is something along the lines of "...assigned a percentile based on the proportion of sites that have a lower diversity."  (then we also wouldn't need to convert it to a percentile later).
%@ELD: It originally said percentile, but Josh seemed to find that to be a more confusing term than "rank", so I switched. I agree that percentile makes it clearer how ties are handled, though, so I'll switch back.

\begin{figure}[t]
\begin{center}
%\includegraphics[width=2.1in]{50065_all_hotspots.png}
\includegraphics[width=2.8in]{figs/localdiversity.png}
\caption{Quantile-quantile plot comparing the distribution of diversity ranks observed for the most complex trait (EQU) to a uniform distribution \citep{oldford_qqtest:_2016}. The blue points compare the values of points at the equivalent percentiles of the two distributions. The shaded area represents the region where we would expect these points to fall if these distributions were identical. Since the points are, for the most part, within this area, we conclude that the distribution of ranks does not deviate substantially from a uniform distribution. This finding suggests that there is nothing special about the diversity in the regions where traits are more likely to evolve, relative to the diversity in the rest of the environment. Results for other complex traits are similar.}
\label{diversity}
\end{center}
\end{figure}

%Is this big enough?
% @CAO: Not really.  Ideally, we should make it two columns, or at least only two populations wide and take up one full column.  I think we have the room (and if it goes only a little over, we can cut)
%This figure could either compare across tasks or across environments. I'm not sure which is better.. Currently it's EQU across all environments. I think I like it better than tasks within an environment, which is what I did for the poster.
% @CAO: I like it how it is.
%I should probably either label the panels A-I, or put some sort of identifier below each.
%ELD: How about now? I could make it wider, but then it would be so tall that it would take up its own page and we probably don't have space for that
\begin{figure*}[t]
\begin{center}
%\includegraphics[width=2.1in]{50065_all_hotspots.png}
\includegraphics[width=5in]{figs/control_9_paths.png}
\caption{Spatial paths taken by the first lineages to evolve the most complex trait (EQU) from five arbitrarily-selected replicates from each environment. The path from each replicate is colored a different shade of gray. As in Figure \ref{hotspots}, background colors represent the set of resources in each part of the environment. I) shows the homogeneous control environment.}
\label{paths}
\end{center}
\end{figure*}

Next, we located the cell in which a novel trait was about to evolve and recorded its percentile. If diversity has no impact on the probability of a new trait evolving, we would expect all percentile values to be equally likely (i.e. follow a uniform distribution). If, on the other hand, new traits are more likely to evolve in areas with elevated local diversity, we would expect high percentiles to be overrepresented.

To test this hypothesis, we compared the distribution of observed percentiles to a uniform distribution. For some of the simple traits, low percentiles were overrepresented due to the presence of populations that contained large swaths of area populated by minimally evolved organisms with none of the traits. However, among the more complex traits, we found no substantial deviation from a uniform distribution (see Figure \ref{diversity}). This result suggests that traits did not evolve more frequently in the most diverse regions of the environment. While it is possible that this lack of an effect is the result of the specific diversity metric used, it strongly suggests that local diversity is not an important driver of hotspot location. This result may also cast doubt on the idea that hotspots of diversity are hotspots of all evolutionary processes, but we would need to perform the same test with phylogenetic rather than phenotypic diversity to really address that point.

%If this section is confusing, I could easily add a heat map of local diversity to help clarify what I'm talking about.
% @CAO: I think it's relatively clear -- I'm not sure a heat map would clear it up much, but if it looks good we can do that.  It does look like we have the space.
% @ELD: Eh, they don't look as cool as you would hope. And I'm thinking that after the figure 5 resize we don't really have the space.

%Thinking about it more, I don't think comparing to the control actually makes sense here. If there were an effect of diversity, it's entirely possible that it would show up in the control too. The control is just unlikely to have high levels of diversity in a consistent region, because there is nothing consistent about specific regions between runs.
% @CAO: I agree.  I think we're doing a good comparison already.  The other issue, though, is that if the natural grouping of sites into a single percentile is driving some of the effect.  The way we control for THAT is to use the same distribution of diversities in the world, but draw from them randomly as our control data.  If we systematically see more or less of an effect, that means diversity did have some kind of correlation that we were missing.
%@ELD: Ah, yes, good point. Preliminarily, it doesn't look like that's happening for the more complex task (it only really happens with 0s), but this is a good fix.


A third potential driver of the evolutionary hotspots is the trajectories that lineages take through space over their lifetimes. Each potential sequence of cells that a lineage goes through represents a different sequence of environments. Perhaps some sequences of environments are more conducive to the evolution of certain traits than others are. To explore this idea further, we located the first organism with each trait across all replicates and traced their lineages back, noting the spatial location of each ancestor. We then overlaid these paths on the corresponding environment (see Figure \ref{paths}). While a thorough statistical analysis of the variation in paths across different traits and environments is beyond the scope of this paper, we do see some qualitative patterns. Of course, the mere presence of hotspots enforces some commonality on the trajectories of lineages that evolved the same trait in the same environment, as a disproportionate number of them will end in the same general area. However, some traits display consistent patterns across the middle portion of their associated paths, which are not seen in paths generated from the control data. For instance, in the three environments in which most of the hotspots are overlapping and in the center (D, E, and G in Figure \ref{paths}), it is rare for any lineage that has ever gone through the lower right corner to be the first one to evolve the most complex task (EQU). This pattern is almost certainly related to the fact that all replicates are seeded with a single organism in the upper left corner; in these environments, successful lineages go directly to the hotspots and don't leave. In other environments, however, successful lineages appear to travel through different sequences of regions (see Figure \ref{paths}F, as an example). While more research will be required to understand the mechanisms behind such patterns, the cumulative effect of lineage trajectory currently seems to be the most promising explanation for the existence of evolutionary hotspots.

%Should I point out the thing about hotspots tending to be in the magenta areas? It's super handwavy, but it's also a thing other people might notice and ask about.

\section{Conclusions}
    We have demonstrated the existence of spatial variation in evolutionary potential across a heterogeneous environment. Research on what specific aspects of a local region are responsible for elevated evolutionary potential is ongoing. The failure of obvious explanations, such as resource presence/absence and local diversity, to predict the location of hotspots suggests that the actual drivers are complex and nuanced. Preliminary investigations of the sequence of locations that a lineage passes through suggest that this avenue is promising for further exploration; there appear to be broad commonalities among paths of lineages that evolved the same trait in the same environment.
 
    Understanding the drivers of spatial variation in evolutionary potential would give us a valuable tool for controlling evolutionary trajectories. The most immediate applications are in evolutionary computation. Many evolutionary algorithms vary the fitness landscape over time, starting with easy problems and working up to more challenging problems \citep{hornby_alps:_2006,ovaska_periodical_2009}. This approach often requires hand-tuning of the different problems and when they appear in the environment. Varying the fitness landscape across space instead could achieve the same outcome. If we can purposefully build spatial networks of fitness landscapes to promote the evolution of a solution to the problem, that would make such an approach even more powerful.  
    
    In the longer term, once our findings have been replicated in a biological system, we hope to use these same concepts to promote the evolution of biological traits of interest. For example, we may be able to engineer habitats that increase the chances of species adapting to climate change through evolutionary rescue. Conversely, we may also be able to arrange environments to inhibit the evolution of undesirable traits, such as antibiotic resistance in bacteria. Overall, this line of research presents a variety of promising applications. 


\section{Acknowledgements}

We extend our thanks to Lindsay Williams for her help with an early version of this manuscript, and to Ashton Shortridge for suggesting the kernel density map method of locating hotspots. We also thank Michael Wiser, Alexander Lalejini, and Rosangela Canino-Koning for comments on the manuscript. This research has been supported in part by the National Science Foundation (NSF) BEACON Center under Cooperative Agreement DBI-0939454, by a National Science Foundation Graduate Research Fellowship to ED under Grant No. DGE-1424871, and by Michigan State University through computational resources provided by the Institute for Cyber-Enabled Research. Any opinions, findings, and conclusions or recommendations expressed in this material are those of the author(s) and do not necessarily reflect the views of the NSF.


\footnotesize


